source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -127,6 +127,8 @@ Additionally, Drivers that implement a Connection Pool MUST support the followin
 
     interface ConnectionPoolOptions {
       /**
+       *  NOTE: This option has been deprecated in favor of timeoutMS.
```

Comment:
Ah, for `WaitQueueTimeoutError` not `waitQueueTimeoutMS`. There are two places: the code snippet under `[Checking Out a Connection](https://github.com/mongodb/specifications/blob/master/source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst#checking-out-a-connection):
```
except timeout:
  emit ConnectionCheckOutFailedEvent(reason="timeout")
  throw WaitQueueTimeoutError
```
and under [Connection Pool Errors](https://github.com/mongodb/specifications/blob/master/source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst#connection-pool-errors):
```
/**
 *  Thrown when a driver times out when attempting to check out
 *  a Connection from a Pool
 */
interface WaitQueueTimeoutError {
  message: 'Timed out while checking out a Connection from connection pool';
  address: <pool address>;
}
```
. I was wondering if you think I'd be worthwhile to add a note since these will need to be transformed into timeoutMS errors.

Reply:
I opted not to talk about error transformations in individual specs because we'd have to do it in a lot of them (e.g. server selection, OCSP, etc) and I thought it would make more sense to consolidate that information in the new timeouts spec.

Reply:
Gotcha ya.